k8s-operator/sessionrecording/tsrecorder/tsrecorder.go

```diff
@@ -55,14 +58,54 @@ func (rec *Client) Write(p []byte) (err error) {
 	}
 	j, err := json.Marshal([]any{
 		rec.clock.Now().Sub(rec.start).Seconds(),
-		"o",
+		outputEventCode,
 		string(p),
 	})
 	if err != nil {
 		return fmt.Errorf("error marhalling payload: %w", err)
 	}
+	return rec.write(j)
+}
+
+// WriteResize writes an asciinema resize message. This can be called if
+// terminal size has changed.
+// https://docs.asciinema.org/manual/asciicast/v2/#r-resize
+func (rec *Client) WriteResize(height, width int) (err error) {
+	const resizeEventCode = "r"
+	p := fmt.Sprintf("%dx%d", height, width)
+	j, err := json.Marshal([]any{
+		rec.clock.Now().Sub(rec.start).Seconds(),
+		resizeEventCode,
+		string(p),
+	})
+	if err != nil {
+		return fmt.Errorf("error marhalling payload: %w", err)
+	}
+	return rec.write(j)
+}
+
+// WriteCastHeaders writes asciinema CastHeader. This must be called once,
+// before any payload is sent to the tsrecorder.
+// https://docs.asciinema.org/manual/asciicast/v2/#header
+func (rec *Client) WriteCastHeader(ch sessionrecording.CastHeader) error {
+	var j []byte
+	j, err := json.Marshal(ch)
+	if err != nil {
+		return fmt.Errorf("error writing CastHeader: %w", err)
+	}
+	j = append(j, '\n')
+	return rec.writeCastLine(j)
+}
+
+func (rec *Client) write(j []byte) error {
```

Comment:
I don't think that we should name a function that sets the failed state and does other stuff to `writeCastLine`, the current `writeCastLine` actually takes cast line and writes it, so that name seems more fitting to me

It makes sense to pull the json marshalling down though 👍🏼